### PR TITLE
Revise linker script for CMSIS

### PR DESCRIPTION
**Pull request by lucas-camp**
This splits the linker script for CMSIS and adds a linker script with high memory configuration.

### CODE REVIEW

File: build_tools/stm32f4xx-cmsis-base.ld [hunk 1]
```diff
@@ -0,0 +1,313 @@
+/******************************************************************************
+ * @file     stm32f407.ld
```
Comment (marbre) [inline build_tools/stm32f4xx-cmsis-base.ld:2]:
```suggestion
 * @file     stm32f4xx-cmsis-base.ld
```

File: build_tools/stm32f4xx-cmsis-base.ld [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+ * @brief    GNU Linker Script for Cortex-M based device
+ * @version  V2.2.0
+ * @date     20. January 2021
+ ******************************************************************************/
+/*
+ * Copyright (c) 2009-2021 Arm Limited. All rights reserved.
```
Comment (marbre) [inline build_tools/stm32f4xx-cmsis-base.ld:8]:
```suggestion
 * Copyright (c) 2009-2021 Arm Limited. All rights reserved.
 * Copyright (c) 2021 Fraunhofer-Gesellschaft zur Förderung der angewandten Forschung e.V.
```

File: build_tools/stm32f4xx-highmem-cmsis.ld [hunk 3]
```diff
@@ -0,0 +1,72 @@
+/******************************************************************************
+ * @file     stm32f407.ld
+ * @brief    GNU Linker Script for Cortex-M based device
+ * @version  V2.2.0
+ * @date     20. January 2021
+ ******************************************************************************/
+/*
+ * Copyright (c) 2009-2021 Arm Limited. All rights reserved.
```
Comment (marbre) [inline build_tools/stm32f4xx-highmem-cmsis.ld:8]:
```suggestion
 * Copyright (c) 2009-2021 Arm Limited. All rights reserved.
 * Copyright (c) 2021 Fraunhofer-Gesellschaft zur Förderung der angewandten Forschung e.V.
```

File: build_tools/stm32f4xx-highmem-cmsis.ld [hunk 4]
```diff
@@ -0,0 +1,72 @@
+/******************************************************************************
+ * @file     stm32f407.ld
```
Comment (marbre) [inline build_tools/stm32f4xx-highmem-cmsis.ld:2]:
```suggestion
 * @file     stm32f4xx-highmem-cmsis.ld
```

File: build_tools/third_party/cmsis_device_f4/CMakeLists.txt [hunk 5]
```diff
@@ -19,3 +19,6 @@ target_include_directories(cmsis_device_f4
 )
 
 target_compile_definitions(cmsis_device_f4 PUBLIC -DSTM32F407xx)
+
+# allow ld to find generic linker script
```
Comment (marbre) [inline build_tools/third_party/cmsis_device_f4/CMakeLists.txt:23]:
```suggestion
# Allow ld to find generic linker script
```

**Review by marbre — CHANGES_REQUESTED** [review]
Thanks for the work. Some minor remarks.

File: build_tools/README.md [hunk 6]
```diff
@@ -3,4 +3,5 @@
...
 The `stm32f4xx-highmem-libopencm3.ld` is a linker script adopted to a custom memory configuration.
 
-The [`stm32f407-cmsis.ld`](https://github.com/ARM-software/CMSIS_5/blob/ca1b514243d8e69f1a8190e59de4b0c4ea6bdcaa/Device/_Template_Vendor/Vendor/Device/Source/GCC/gcc_arm.ld) (renamed from gcc_arm.ld) linker script was imported with minor modifications from the [ARM-software/CMSIS_5](https://github.com/ARM-software/CMSIS_5) repository.
+The `stm32f4xx-cmsis-base.ld` and `stm32f407-cmsis.ld` linker scripts are based on the the [`gcc_arm.ld`](https://github.com/ARM-software/CMSIS_5/blob/ca1b514243d8e69f1a8190e59de4b0c4ea6bdcaa/Device/_Template_Vendor/Vendor/Device/Source/GCC/gcc_arm.ld) linker script from the [ARM-software/CMSIS_5](https://github.com/ARM-software/CMSIS_5) repository. The common parts were extracted into `stm32f4xx-cmsis-base.ld` and the memory layout definition into `stm32f407-cmsis.ld`.
```
Comment (marbre) [inline build_tools/README.md:6]:
```suggestion
The `stm32f4xx-cmsis-base.ld` and `stm32f407-cmsis.ld` linker scripts are based on the the [`gcc_arm.ld`](https://github.com/ARM-software/CMSIS_5/blob/ca1b514243d8e69f1a8190e59de4b0c4ea6bdcaa/Device/_Template_Vendor/Vendor/Device/Source/GCC/gcc_arm.ld) linker script from the [ARM-software/CMSIS_5](https://github.com/ARM-software/CMSIS_5) repository. The `gcc_arm.ld` linker script was split, such that common parts are in `stm32f4xx-cmsis-base.ld` whereas the memory layout is defined in `stm32f407-cmsis.ld`.
```

Comment (lucas-camp) [reply build_tools/README.md]: Should be fixed now.

**Comment by marbre** [comment]
Forgot to approve before merging, but looks good to me!

**Pull request merged**